src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+import duke.tasks.Task;
+
+/**
+ * List data structure to store the tasks in Duke
+ */
+public class TaskList implements Serializable {
+
+    private static final String INVALID_INDEX_EXCEPTION = "Invalid task index given for Mark/Unmark/Delete command.";
+    private static final String ADDED_TASK_RESPONSE = "Task added:\n";
+    private static final String REMAINING_TASK_RESPONSE = "\nRemaining task count: ";
+    private static final String REMOVE_TASK_RESPONSE = "Task removed:\n";
+    private static final String LIST_INDEX_SEPARATOR = ". ";
+
+    private ArrayList<Task> list;
+    public TaskList() {
+        list = new ArrayList<>();
```

Comment:
Just nitpicking a little but could name this ArrayList variable more comprehensively e.g. listOfTasks

Reply:
Right, that's a good catch. Thanks!